src/phoenix/trace/dsl/filter.py

```diff
@@ -18,6 +23,60 @@
 )
 
 
+EvalAttribute: TypeAlias = typing.Literal["label", "score"]
+EvalExpression: TypeAlias = str
+EvalName: TypeAlias = str
+
+
+@dataclass(frozen=True)
+class AliasedAnnotationRelation:
+    """
+    Represents an aliased `span_annotation` relation (i.e., SQL table). Used to
+    perform joins on span evaluations during filtering. An alias is required
+    because the `span_annotation` may be joined multiple times for different
+    evaluation names.
+    """
+
+    index: int
+    name: str
+    AliasedSpanAnnotation: AliasedClass[models.SpanAnnotation] = field(init=False, repr=False)
+    _label_attribute_alias: str = field(init=False, repr=False)
+    _score_attribute_alias: str = field(init=False, repr=False)
+
+    def __post_init__(self) -> None:
+        table_alias = f"span_annotation_{self.index}"
+        alias_id = f"{randint(0, 10**6):06d}"  # prevent conflicts with user-defined attributes
+        label_attribute_alias = f"{table_alias}_label_{alias_id}"
+        score_attribute_alias = f"{table_alias}_score_{alias_id}"
+        AliasedSpanAnnotation = aliased(models.SpanAnnotation, name=table_alias)
+        object.__setattr__(self, "_label_attribute_alias", label_attribute_alias)
+        object.__setattr__(self, "_score_attribute_alias", score_attribute_alias)
+        object.__setattr__(
+            self,
+            "AliasedSpanAnnotation",
```

Comment:
We can just call this `_table`, which is probably easier to understand.
```suggestion
            "_table",
```